components/tyk-mdcb/templates/deployment-mdcb.yaml

```diff
@@ -0,0 +1,144 @@
+          image: "{{ .Values.mdcb.image.repository }}:{{ .Values.mdcb.image.tag }}"
+          imagePullPolicy: {{ .Values.mdcb.image.pullPolicy }}
+          command: ["/opt/tyk-sink/tyk-sink", "--c=/etc/tyk-sink/tyk_sink.conf"]
+          ports:
+            - containerPort: {{ .Values.mdcb.listenPort }}
+            - containerPort: {{ .Values.mdcb.probes.healthCheckPort }}
+          livenessProbe:
+            httpGet:
+              path: {{ .Values.mdcb.probes.liveness.path }}
+              port: {{ .Values.mdcb.probes.healthCheckPort}}
+              scheme: {{ .Values.mdcb.probes.liveness.scheme }}
+            initialDelaySeconds: {{ .Values.mdcb.probes.liveness.initialDelaySeconds }}
+            periodSeconds: {{ .Values.mdcb.probes.liveness.periodSeconds }}
+            timeoutSeconds: {{ .Values.mdcb.probes.liveness.timeoutSeconds }}
+            failureThreshold: {{ .Values.mdcb.probes.liveness.failureThreshold }}
+          readinessProbe:
+            initialDelaySeconds: {{ .Values.mdcb.probes.readiness.initialDelaySeconds }}
+            periodSeconds: {{ .Values.mdcb.probes.readiness.periodSeconds }}
+            timeoutSeconds: {{ .Values.mdcb.probes.readiness.timeoutSeconds }}
+            failureThreshold: {{ .Values.mdcb.probes.readiness.failureThreshold }}
+            httpGet:
+              path: {{ .Values.mdcb.probes.readiness.path }}
+              port: {{ .Values.mdcb.probes.healthCheckPort}}
+              scheme: {{ .Values.mdcb.probes.liveness.scheme }}
+          resources:
+            {{- toYaml .Values.mdcb.resources | nindent 12 }}
+          volumeMounts:
+            - name: mdcb-conf
+              mountPath: /etc/tyk-sink
+              {{- if .Values.mdcb.extraVolumeMounts }}
+              {{- include "mdcb.tplvalues.render" (dict "value" .Values.mdcb.extraVolumeMounts "context" $) | nindent 10 }}
+              {{- end }}
+          env:
+            - name: TYK_MDCB_LISTENPORT
+              value: "{{ .Values.mdcb.listenPort }}"
+            - name: TYK_MDCB_HEALTHCHECKPORT
+              value: "{{ .Values.mdcb.probes.healthCheckPort }}"
+            - name: TYK_MDCB_STORAGE_TYPE
+              value: "redis"
+            - name: TYK_MDCB_STORAGE_ENABLECLUSTER
+              value: "{{ default "false" .Values.global.redis.enableCluster }}"
+            - name: TYK_MDCB_STORAGE_ADDRS
+              value: {{ include "mdcb.redis_url" . | quote }}
+            - name: TYK_MDCB_STORAGE_PORT
+              value: "{{ .Values.mdcb.probes.healthCheckPort }}"
+            - name: TYK_MDCB_HASHKEYS
+              value: "{{ .Values.mdcb.hashKeys }}"
+            - name: TYK_MDCB_STORAGE_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "mdcb.redis_secret_name" . }}
+                  key: {{ include "mdcb.redis_secret_key" . }}
+            - name: TYK_MDCB_STORAGE_REDISUSESSL
+              value: "{{ default "false" .Values.global.redis.useSSL }}"
```

Comment:
we might need default here since `redis.useSSL` field is not defined and will be defined by user. i mean, `redis.useSSL` field is commented in values.yaml file. so, there is a high possibility for users to leave it empty